packages/components/pie-button/test/helpers/page-object/pie-button-form-integration.page.ts

```diff
@@ -0,0 +1,188 @@
+    private readonly userPaymentCardTypeSelect: Locator;
+    private readonly userPaymentCardNumberInput: Locator;
+    private readonly userPaymentCardExpirationInput: Locator;
+    private readonly resetPieButton: Locator;
+    private readonly resetNativeButton: Locator;
+    private readonly submitPieButton: Locator;
+    readonly formSubmittedFlag: Locator;
+
+    constructor (page: Page, componentName: string) {
+        super(page, componentName);
+
+        this.formLocator = page.getByTestId(form.selectors.container.dataTestId);
+        this.userNameInput = this.formLocator.getByTestId(form.selectors.name.dataTestId);
+        this.userEmailInput = this.formLocator.getByTestId(form.selectors.email.dataTestId);
+        this.userPasswordInput = this.formLocator.getByTestId(form.selectors.password.dataTestId);
+        this.userPaymentCardTypeSelect = this.formLocator.getByTestId(form.selectors.userPaymentCardType.dataTestId);
+        this.userPaymentCardNumberInput = this.formLocator.getByTestId(form.selectors.userPaymentCardNumber.dataTestId);
+        this.userPaymentCardExpirationInput = this.formLocator.getByTestId(form.selectors.userPaymentCardExpiration.dataTestId);
+
+        this.resetPieButton = this.formLocator.locator(`[data-test-id="${form.selectors.pieButtonReset.dataTestId}"]`);
+        this.resetNativeButton = this.formLocator.locator(`[data-test-id="${form.selectors.nativeButtonReset.dataTestId}"]`);
+        this.submitPieButton = this.formLocator.locator(`[data-test-id="${form.selectors.pieButtonSubmit.dataTestId}"]`);
+
+        this.formSubmittedFlag = this.page.getByTestId(form.selectors.formSubmittedFlag.dataTestId);
+    }
+
+    /**
+     * Fills out the form with the provided input data.
+     *
+     * @param {FormInput} inputData - The data to fill the form with.
+     * @returns {Promise<FormIntegrationPage>} A Promise that resolves with the current instance once the form has been filled.
+     */
+    async fillForm (inputData: FormInput): Promise<FormIntegrationPage> {
+        await this.userNameInput.fill(inputData.userName);
+        await this.userEmailInput.fill(inputData.userEmail);
+        await this.userPasswordInput.fill(inputData.userPassword);
+        await this.userPaymentCardTypeSelect.selectOption(inputData.userPaymentCardType);
+        await this.userPaymentCardNumberInput.fill(inputData.userPaymentCardNumber);
+        await this.userPaymentCardExpirationInput.fill(inputData.userPaymentCardExpiration);
+
+        return this;
+    }
+
+    /**
+     * Clicks the specified form button.
+     *
+     * @param {FormButtonType} buttonType - The type of the button to click.
+     * @returns {Promise<void>} A Promise that resolves once the button has been clicked.
+     */
+    async clickFormButton (buttonType: FormButtonType): Promise<void> {
+        switch (buttonType) {
+            case 'pie-button-submit':
+                await this.submitPieButton.click();
```

Comment:
I think we could remove `async/await` and just return `this.submitPieButton.click();` as that's the promise consumers can await on their end. But I could be mistaken

Reply:
Will be removing this as there's a `clickButtonWithText` function I didn't see in the base page object